### ISSUE TITLE
5.4 field duplication?

### ISSUE THREAD

**Issue by jonathanstegall** [helgatheviking/Nav-Menu-Roles#40]
It looks like you've been involved in the 5.4 work to get some menu hooks into Core (which is wonderful). I notice, when running `5.4-RC4-47505` in the beta plugin, that the fields for this plugin are duplicated.  There's a [ticket](https://core.trac.wordpress.org/ticket/49500) about this that talks about version checking, but looking through this plugin's code I'm not entirely sure where to put that.

Do you already have plans to address this before 5.4 is released? I can do a bit more digging if not, but figured I'd check in first.

Edit to include what I tried already: Putting the version check in `edit_nav_menu_walker` where I thought it would go doesn't seem to have an effect.

**Comment by helgatheviking** [comment]
I started something locally but didn't finish.... I have to prioritize paying projects so this doesn't always get the most attention. I've just pushed up now

https://github.com/helgatheviking/Nav-Menu-Roles/tree/issues/40-5.4-compat 

It would be a big help if you could test it out. though I think you may need to tweak [this](https://github.com/helgatheviking/Nav-Menu-Roles/blob/issues/40-5.4-compat/nav-menu-roles.php#L112) to be `5.4-RC4` instead of `5.4` strict. 

Not sure what else is required, but for sure we need to version test WP and only add [this](https://github.com/helgatheviking/Nav-Menu-Roles/blob/master/nav-menu-roles.php#L112) if it's less than WP 5.4.... which hopefully the new branch resolves.

**Comment by jonathanstegall** [comment]
Your changes make it work perfectly for me in 5.4, as long as the version check is correctly running. Presumably this won't be necessary once it's not an RC, but I borrowed this from an older Gutenberg commit.

```php
public static function is_wp_gte( $version = '5.4.0' ) {
	global $wp_version;
	return version_compare( strtok( $wp_version, '-' ), $version, '>=' ) ? true : false;
}
```

**Comment by helgatheviking** [comment]
Hmmmm... the concept is right but version checking the RC seems dodgy. If I force `$wp_version` to 5.4 then seems to work.

**Comment by jonathanstegall** [comment]
That would make sense. Thank you!

**Comment by helgatheviking** [comment]
I'm seeing that dropping the trailing 0 off the string I'm trying to compare is working.

```
if( ! self::is_wp_gte( '5.4-RC1' ) ) {
			add_filter( 'wp_edit_nav_menu_walker', array( $this, 'edit_nav_menu_walker' ) );
		}
```

and then later

```
	public static function is_wp_gte( $version = '5.4' ) {
		global $wp_version;
		return version_compare( strtolower( $wp_version ), strtolower( $version ), '>=' );
	}
```

Could you confirm that works for you too?

**Comment by jonathanstegall** [comment]
This still doesn't work for me. That is, the version checking does work, but I still see the duplicate fields.

**Comment by helgatheviking** [comment]
Hmmm... do you have any other plugins that might be adding that hook too?

**Comment by jonathanstegall** [comment]
Ugh, I found one. Popup-maker seems to use it. I can disable that though, and when I do, it no longer duplicates. So that all works for me.

**Comment by helgatheviking** [comment]
closed by #41
